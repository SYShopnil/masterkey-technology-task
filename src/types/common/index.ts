export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EDataTestId {
  CAlphabetApp = "CAlphabetApp",
  CAlphabetGrid = "CAlphabetGrid",
  CLayout = "CLayout",
  CPartition = "CPartition",
  SHeaderMain = "SHeaderMain",
  SIconWithMessage = "SIconWithMessage",
  SLoading = "SLoading",
}
