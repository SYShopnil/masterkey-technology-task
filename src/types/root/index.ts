export type { IIconStore } from "./_icon";
export { IconName as IconEnums } from "./_icon";
