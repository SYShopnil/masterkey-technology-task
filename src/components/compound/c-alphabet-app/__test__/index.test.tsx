import { render, screen } from "@testing-library/react";
import { EDataTestId } from "@src/types/common";
import { CAlphabetApp } from "..";

describe("When Alphabet Component Rendered", () => {
  it("Expect render successfully", () => {
    render(<CAlphabetApp />);
    const myElement = screen.getByRole(EDataTestId.CAlphabetApp);
    expect(myElement).toMatchSnapshot();
  });
});
