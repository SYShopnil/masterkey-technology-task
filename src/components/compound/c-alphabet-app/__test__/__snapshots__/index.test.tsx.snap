// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Alphabet Component Rendered Expect render successfully 1`] = `
<div
  class="p-4"
  role="CAlphabetApp"
>
  <div>
    <h1
      class="text-center text-2xl font-bold pb-2"
    >
      Alphabet Tile Interaction
    </h1>
    <hr />
  </div>
  <div>
    <div
      class="mb-4 text-xl font-bold"
      id="outputString"
    />
    <div
      class="grid grid-cols-6 gap-2 p-4"
      role="CAlphabetGrid"
    >
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        A
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        B
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        C
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        D
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        E
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        F
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        G
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        H
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        I
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        J
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        K
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        L
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        M
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        N
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        O
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        P
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        Q
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        R
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        S
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        T
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        U
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        V
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        W
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        X
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        Y
      </div>
      <div
        class="flex items-center justify-center w-12 h-12 m-1 text-2xl font-bold bg-blue-500 text-white cursor-pointer"
      >
        Z
      </div>
    </div>
  </div>
</div>
`;
